lib/binding/kvdb_interface.c

```diff
@@ -192,23 +195,25 @@ rest_global_params_get(
 
         buf = malloc(buf_sz * sizeof(*buf));
         if (ev(!buf))
-            return REST_STATUS_INTERNAL_SERVER_ERROR;
+            return rest_response_perror(resp, REST_STATUS_SERVICE_UNAVAILABLE, "Out of memory");
```

Comment:
Can you just use a small, fixed sized buffer to start, then malloc a bigger one if necessary?  Seems like it would really simplify this code...

Reply:
How about I come back through here on a follow-up PR?